agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/statsbeat/NetworkStatsbeat.java

```diff
@@ -178,51 +191,60 @@ private void sendIntervalMetric(
-      addCommonProperties(requestFailureCountSt, ikey, local.host);
+      addCommonProperties(requestFailureCountSt, ikey, local.host, local.cause);
       telemetryClient.trackStatsbeatAsync(requestFailureCountSt.build());
     }
 
     double durationAvg = local.getRequestDurationAvg();
     if (durationAvg != 0) {
       StatsbeatTelemetryBuilder requestDurationSt =
           createStatsbeatTelemetry(telemetryClient, REQUEST_DURATION_METRIC_NAME, durationAvg);
-      addCommonProperties(requestDurationSt, ikey, local.host);
+      addCommonProperties(requestDurationSt, ikey, local.host, local.cause);
       telemetryClient.trackStatsbeatAsync(requestDurationSt.build());
     }
 
     if (local.retryCount.get() != 0) {
       StatsbeatTelemetryBuilder retryCountSt =
           createStatsbeatTelemetry(
               telemetryClient, RETRY_COUNT_METRIC_NAME, (double) local.retryCount.get());
-      addCommonProperties(retryCountSt, ikey, local.host);
+      addCommonProperties(retryCountSt, ikey, local.host, local.cause);
       telemetryClient.trackStatsbeatAsync(retryCountSt.build());
     }
 
     if (local.throttlingCount.get() != 0) {
       StatsbeatTelemetryBuilder throttleCountSt =
           createStatsbeatTelemetry(
               telemetryClient, THROTTLE_COUNT_METRIC_NAME, (double) local.throttlingCount.get());
-      addCommonProperties(throttleCountSt, ikey, local.host);
+      addCommonProperties(throttleCountSt, ikey, local.host, local.cause);
       telemetryClient.trackStatsbeatAsync(throttleCountSt.build());
     }
 
     if (local.exceptionCount.get() != 0) {
       StatsbeatTelemetryBuilder exceptionCountSt =
           createStatsbeatTelemetry(
               telemetryClient, EXCEPTION_COUNT_METRIC_NAME, (double) local.exceptionCount.get());
-      addCommonProperties(exceptionCountSt, ikey, local.host);
+      addCommonProperties(exceptionCountSt, ikey, local.host, local.cause);
       telemetryClient.trackStatsbeatAsync(exceptionCountSt.build());
     }
   }
 
-  private static void addCommonProperties(
-      StatsbeatTelemetryBuilder telemetryBuilder, String ikey, String host) {
+  private static <T> void addCommonProperties(
+      StatsbeatTelemetryBuilder telemetryBuilder, String ikey, String host, @Nullable T cause) {
     telemetryBuilder.addProperty("endpoint", BREEZE_ENDPOINT);
     telemetryBuilder.addProperty("cikey", ikey);
     telemetryBuilder.addProperty("host", host);
+    if (cause != null) {
+      // track 'statusCode' for Failure/Retry/Throttle count and track 'exceptionType' for Exception
+      // Count
+      if (cause instanceof Integer) {
+        telemetryBuilder.addProperty("statusCode", ((Integer) cause).toString());
+      } else if (cause instanceof String) {
+        telemetryBuilder.addProperty("exceptionType", (String) cause);
+      }
+    }
   }
 
-  private static class IntervalMetrics {
+  private static class IntervalMetrics<T> {
```

Comment:
I'm not sure about using generic for this, but thing may change after addressing the possible issue with `cause` getting overwritten, so let's revisit after that